src/control/server/drpc.go

```diff
@@ -162,15 +162,17 @@ func newDrpcCall(module int32, method int32, bodyMessage proto.Message) (*drpc.C
 // makeDrpcCall opens a drpc connection, sends a message with the
 // protobuf message marshalled in the body, and closes the connection.
 // drpc response is returned after basic checks.
-func makeDrpcCall(
-	client drpc.DomainSocketClient, module int32, method int32,
+func makeDrpcCall(client drpc.DomainSocketClient, module int32, method int32,
 	body proto.Message) (drpcResp *drpc.Response, err error) {
 
 	drpcCall, err := newDrpcCall(module, method, body)
 	if err != nil {
 		return drpcResp, errors.Wrap(err, "build drpc call")
 	}
 
+	client.Lock()
+	defer client.Unlock()
```

Comment:
is this recommended practice? I understand you'd have to unlock in 4 places if you didn't do this but I always worry about relying on someone else to implicitly unlock a mutex. If this is how they recommend doing it I guess its ok.

Reply:
I'm not sure I understand the concern, the mutex on the socket client will be unlocked when makeDrpcCall returns. we don't really mean to hold the lock during checkDrpcResponse so I can unlock before then if we think that is significant.